src/main/java/seedu/condonery/logic/commands/property/EditPropertyCommand.java

```diff
@@ -110,11 +112,14 @@ public CommandResult execute(Model model) throws CommandException {
             throw new CommandException(MESSAGE_DUPLICATE_PROPERTY);
         }
 
-        File existingImage = new File(propertyToEdit.getImagePath().toString());
-        if (existingImage.exists()) {
-            existingImage.renameTo(new File(editedProperty.getImagePath().toString()));
+        if (propertyToEdit.getImagePath() != null) {
```

Comment:
not sure if i agree with this checking, since getImagePath should never be null. Doing this might prevent errors but is more likely burying the issue. 

The main problem is we need to be careful whenever creating Properties that they are created with the correct imageDirectoryPath

Reply:
Fixed in the latest commit !